firestore/src/include/firebase/firestore/local_cache_settings.h

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_FIRESTORE_SRC_INCLUDE_FIREBASE_FIRESTORE_LOCAL_CACHE_SETTINGS_H_
+#define FIREBASE_FIRESTORE_SRC_INCLUDE_FIREBASE_FIRESTORE_LOCAL_CACHE_SETTINGS_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "Firestore/core/src/api/settings.h"
```

Comment:
I don't think we can include from the ios sdk in the "include" directory because this include also has to work with android. You may have to use a pimpl style like how `firestore/src/include/firebase/firestore/document_reference.h` does using `DocumentReferenceInternal`.

Reply:
There are still includes from the "main" directory in `local_cache_settings.h`. This is fine for now but will fall apart once you implement android support. The APIs published in the "include" directory must be agnostic of the main/android implementation.

Reply:
Remove all traces of the strings "main" and "api::" from `local_cache_settings.h`. It still has this line:

```
#include "firestore/src/main/local_cache_settings_main.h"
```

which won't work when targeting Android.